### ISSUE TITLE
Replace `Invocation` struct with protobuf

### ISSUE THREAD

**Issue by tiziano88** [project-oak/oak#1249]
https://github.com/project-oak/oak/blob/7b89fbdada697230672eae48b10fbe7904a33b2f/sdk/rust/oak/src/grpc/invocation.rs#L17-L22

Now that we have support for handles in protobuf, we should replace data structures that we currently manually serialize, with alternative protobuf versions.

@conradgrobler would you like to look into it, since we talked about this before?

**Comment by conradgrobler** [comment]
There is an `Invocation` in the SDK (for use by WASM nodes) and one in the Runtime (for use by the gRPC pseudo nodes). They depend on slightly different versions of other structs and implement slightly different but similar traits. I think #1224 is a prerequisite for this.

**Comment by conradgrobler** [comment]
As a first step, a pragmatic version of this is needed as part of #1246. Instead of trying to replace both instances of `Invocation` with a single proto, the first step (that will be done as part of #1246) is to only replace the SDK `Invocation`. Currently support for handles in protos is only available in WASM nodes that depend on the SDK.

In the runtime, the current implementations of `Encodable` and `Decodable` for `Invocation` will be modified to ensure that it is compatible with the messages generated from the `Invocation' proto in the SDK, but will remain as a manually implemented struct.

**Comment by tiziano88** [comment]
Is this still blocked on #1246 ?

**Comment by conradgrobler** [comment]
This is blocked by #1224. The plan was to implement the first part of this (the part not blocked by #1224) as part of #1246 as it is required there.

The rest would have to wait until we did some significant refactoring or redesigning. #1224 is problematic. I started working on it, but decided to back out of the changes when I had to move more than half (if I remember correctly) of the code from the SDK to the ABI crate, and many of the structs I had to move did not seem appropriate for the ABI crate. This is because all handle serialisation logic is currently in the SDK and needs to move to the ABI to be reusable from the runtime. But it had many direct and indirect dependencies, so all of those had to be moved as well. Another issue is the Sender and Receiver trait implementations which could not be moved to the ABI. the version in the SDK is intended to run inside a WASM node, where the version in the Runtime calls the runtime functions directly. Code generation is also a problem. Even just the part planned for inclusion in #1246 is already proving difficult and a potential blocker.
